pyresample/gradient/__init__.py

```diff
@@ -33,25 +36,49 @@
 from shapely.geometry import Polygon
 
 from pyresample import CHUNK_SIZE
-from pyresample.geometry import get_geostationary_bounding_box
-from pyresample.gradient._gradient_search import one_step_gradient_search
-from pyresample.resampler import BaseResampler
+from pyresample.geometry import (
+    AreaDefinition,
+    SwathDefinition,
+    get_geostationary_bounding_box,
+)
+from pyresample.gradient._gradient_search import (
+    one_step_gradient_indices,
+    one_step_gradient_search,
+)
+from pyresample.resampler import BaseResampler, resample_blocks
 
 logger = logging.getLogger(__name__)
 
 
+def GradientSearchResampler(source_geo_def, target_geo_def):
+    """Create a gradient search resampler."""
+    warnings.warn("`GradientSearchResampler` is deprecated, please use "
+                  "`create_gradient_search_resampler` instead.",
+                  DeprecationWarning)
```

Comment:
Satpy expects the resampler to be a class, how will the `create_gradient_search_resampler()` be used there?

Reply:
From what I see, it's just name of variable that hints at it being a class, but it can be a function...

Reply:
Oh, indeed. So the current `GradientSearchResampler` just needs to be replaced with `create_gradient_search_resampler()` in some point.